cdap-docs/reference-manual/source/http-restful-api/metrics.rst

```diff
@@ -371,161 +416,201 @@ To query a metric within a given context, perform an HTTP POST request::
+with a JSON list as the request body that enumerates the name and attributes for each
+metric. The format of the request and the JSON body depends on whether the metrics share
+the same context or are being called for different contexts. 
+
+.. rubric:: Multiple Metrics with the Same Context
+
+Retrieving multiple metrics at once for the same contexts can be accomplished by issuing a
+request as in previous examples, but providing the additional metrics. For example::
+
+  POST '<base-url>/metrics/query?tag=flow:CountRandom&metric=system.process.events.processed
+    &metric=system.dataset.store.bytes&start=now-5s&count=5'
+
+The result (pretty-printed to fit) would be::
+
+  {"startTime":1429487786,
+   "endTime":1429487791,
+   "series":[{"metricName":"system.process.events.processed",
+              "grouping":{},
+              "data":[{"time":1429487786,"value":1268},
+                      {"time":1429487787,"value":1324},
+                      {"time":1429487788,"value":1206},
+                      {"time":1429487789,"value":1125},
+                      {"time":1429487790,"value":1035}]},
+             {"metricName":"system.dataset.store.bytes",
+              "grouping":{},
+              "data":[{"time":1429487786,"value":15600},
+                      {"time":1429487787,"value":14998},
+                      {"time":1429487788,"value":13712},
+                      {"time":1429487789,"value":12246},
+                      {"time":1429487790,"value":9924}]
+              }]
+  }
 
-To retrieve multiple metrics at once, instead of a GET, issue an HTTP POST, with a JSON
-list as the request body that enumerates the name and attributes for each metrics. For
-example::
+.. rubric:: Multiple Metrics with Different Contexts
 
-  POST http://<host>:<port>/v2/metrics
+Retrieving multiple metrics at once for different contexts can be accomplished by issuing
+a request with a JSON list as the request body that enumerates the name, attributes and
+context for each metric. Use an HTTP POST request:: 
 
-with the arguments as a JSON string in the body. For example, to retrieve multiple metrics
-using a ``curl`` call::
+  POST 'http://<host>:<port>/v3/metrics/query'
 
-  $ curl -w'\n' -X POST 'http://localhost:10000/v2/metrics' \
-  -H "Content-Type: application/json" \
-  -d '[ "/system/collect.events?aggregate=true", '\
-  '"/system/apps/HelloWorld/process.events.processed?start=1380323712&count=6000" ]'
+with the arguments as a JSON string in the body. The format of the JSON follows this
+structure (pretty-printed)::
+
+  { “query1”: {
+        tags: {“namespace”: “default”, “app”: “PurchaseHistory”}, 
+        metrics: [“metric1”, “metric2”],
+        groupBy: [“app”, “dataset”],
+        timeRange: {“aggregate”: “true”}
+        },
+    “query2”: {
+        tags: {“namespace”: “default”},
+        metrics: [“metric1”, “metric2”],
+        groupBy: [“app”, “dataset”],
+        timeRange: {“start”: “now­2s”, “end”: “now”}
+        }
+  }
+
```

Comment:
I tried that with my example, and got no results. "There's no data aggregated for specified tags to satisfy the query"...do you have an example that works?